File: 2106-find-greatest-common-divisor-of-array/2106-find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int mini = 1001, maxi = 0;
        for(int n : nums){
            mini = min(mini, n);
            maxi = max(maxi,n);
        }
        return gcd(mini,maxi);
    }
};

File: 2106-find-greatest-common-divisor-of-array/2106-find-greatest-common-divisor-of-array.py
from math import gcd
class Solution:
    def findGCD(self, nums: List[int]) -> int:
        maxi = 0
        mini = 1001
        for x in nums:
            maxi = max(maxi,x)
            mini = min(mini,x)
        return gcd(maxi,mini)
        